test/js/unit/TinCan-async.js

```diff
@@ -229,11 +291,139 @@
         );
     };
 
+    doGetStatementWithAttachmentTest = function (v) {
+        asyncTest(
+            "getStatementWithAttachment (async): " + v,
+            function () {
+                var sendResult,
+                    //
+                    // Cloud is lowercasing the mbox value so just use a lowercase one
+                    // and make sure it is unique to prevent merging that was previously
+                    // possible, but leaving the commented version as a marker for something
+                    // that ought to be tested against a 1.0.0 spec
+                    //
+                    //actorMbox = "mailto:TinCanJS-test-TinCan+" + Date.now() + "@tincanapi.com";
+                    actorMbox = "mailto:tincanjs-test-tincan+" + Date.now() + "@tincanapi.com";
+
+                sendResult = session[v].sendStatement(
+                    {
+                        actor: {
+                            mbox: actorMbox
+                        },
+                        verb: {
+                            id: "http://adlnet.gov/expapi/verbs/attempted"
+                        },
+                        target: {
+                            id: "http://tincanapi.com/TinCanJS/Test/TinCan_getStatements/async/" + v
+                        },
+                        attachments: [
+                            {
+                                display: {
+                                    "en-US": "Test Attachment"
+                                },
+                                usageType: "http://id.tincanapi.com/attachment/test-attachment",
+                                content: "test content",
+                                contentType: "text/plain"
+                            }
+                        ]
+                    },
+                    function (results, sentStatement) {
+                        // TODO: need to handle errors?
+
+                        var getResult = session[v].getStatement(
+                            sentStatement.id,
+                            function (err, statement) {
+                                start();
+
+                                // clear the "stored" and "authority" properties since we couldn't have known them ahead of time
+                                // TODO: should we check the authority?
+                                statement.stored = null;
+                                statement.authority = null;
+
+                                deepEqual(err, null, "callback: err argument (" + v + ")");
+                                deepEqual(sentStatement.attachments[0], new TinCan.Attachment(statement.attachments[0]), "callback: statement matches (" + v + ")");
```

Comment:
Shouldn't  the statement coming back already have a `TinCan.Attachment`, why construct it here?